### PR TITLE
fix: renamed folder MainMenu->main-menu and support rest props

### PR DESCRIPTION
**Pull request by ad1992**

### CODE REVIEW

File: src/components/dropdownMenu/DropdownMenuItem.tsx [hunk 1]
```diff
@@ -9,30 +9,27 @@ const DropdownMenuItem = ({
   icon,
   onSelect,
   children,
-  dataTestId,
   shortcut,
   className,
-  style,
-  ariaLabel,
+  "aria-label": ariaLabel,
```
Comment (dwelle) [inline src/components/dropdownMenu/DropdownMenuItem.tsx:14]: redundant

File: src/components/dropdownMenu/DropdownMenuItem.tsx [hunk 2]
```diff
@@ -9,30 +9,27 @@ const DropdownMenuItem = ({
...
-      aria-label={ariaLabel}
+      {...rest}
       onClick={onSelect}
-      data-testid={dataTestId}
-      title={ariaLabel}
       type="button"
       className={getDrodownMenuItemClassName(className)}
-      style={style}
+      aria-label={ariaLabel}
+      title={ariaLabel}
```
[the remark this answers is not in the document]
Comment (ad1992) [reply src/components/dropdownMenu/DropdownMenuItem.tsx]: using `rest['aria-label']` gives ts error and fixing ts error by checking if key is present still fails in script for generating types hence I am destructing `aria-label`

[the remark this answers is not in the document]
Comment (dwelle) [reply src/components/dropdownMenu/DropdownMenuItem.tsx]: I cannot reproduce any of those issues

File: src/components/dropdownMenu/DropdownMenuItem.tsx (hunk 1 above)
Comment (dwelle) [reply src/components/dropdownMenu/DropdownMenuItem.tsx:14]: I meant, it's unnecessary. spreading `{...rest}` will take care of it.

File: src/components/dropdownMenu/DropdownMenuItem.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ad1992) [reply src/components/dropdownMenu/DropdownMenuItem.tsx]: You are right its working, I am sure I might have messed up something earlier due to which it was failing in types generation.

**Review by dwelle — APPROVED** [review]

**Pull request merged**